_source/_posts/2022-05-02-angular-microfrontend-auth.md

```diff
@@ -0,0 +1,443 @@
+```
+
+The defaults are mostly correct, except we have a module, not a component that we want to expose. If you want to expose a component instead, all you'd need to do is update which component to expose. Update the configuration snippet to expose the `ProfileModule` by matching the following code snippet:
+
+```js
+// For remotes (please adjust)
+name: "mfeProfile",
+filename: "remoteEntry.js",
+exposes: {
+  './Module': './projects/mfe-profile/src/app/profile/profile.module.ts',
+},
+```
+
+Now we can incorporate the micro-frontend in the `shell` application. Open `projects/shell/webpack.config.js`. Here is where you'll add the new micro-frontend so that the `shell` application knows how to access it. In the middle of the file, inside the `plugins` array, there's a property for `remotes`. The micro-frontend in the starter code, `mfeBasket`, is already added to the `remotes` object. There is where you'll add the remote for `mfeProfile` following the same pattern but replacing the port to 4202. Update your configuration to look like this.
+
+```js
+// For hosts (please adjust)
+remotes: {
+  "mfeBasket": "http://localhost:4201/remoteEntry.js",
+  "mfeProfile": "http://localhost:4202/remoteEntry.js"
+},
+```
+
+We can add update the code to incorporate the profile's micro-frontend. Open `projects/shell/src/app/app-routing.module.ts`. Add a path to the profile micro-frontend to the routes array using the path 'profile'. Your routes array should look like this.
+
+```ts
+const routes: Routes = [
+  { path: '', component: ProductsComponent },
+  { path: 'basket', loadChildren: () => import('mfeBasket/Module').then(m => m.BasketModule) },
+  { path: 'profile', loadChildren: () => import('mfeProfile/Module').then(m => m.ProfileModule)},
+  { path: 'login/callback', component: OktaCallbackComponent }
+];
+```
+
+What's this!? The IDE flags the import path as an error! The `shell` application code doesn't know about the Profile module, and TypeScript needs a little help. Open `projects/shell/src/decl.d.ts` and add the following line of code.
+
+```ts
+declare module 'mfeProfile/Module';
+```
+
+The IDE should be happier now. 😀
+
+Next, update the navigation button for **Profile** in the `shell` application to route to the correct path. Open `projects/shell/src/app/app.component.html` and find the `routerLink` for the **Profile** button. It should be approximately on line 38. Right now the `routerLink` configuration is `routerLink="/"`, but it should now be
+
+```html
+<a routerLink="/profile">
+```
+
+This is everything we need to do to connect the micro-frontend remote to the host application, but we also want to share authenticated state. Module Federation makes sharing state a piece of (cup) cake.
+
+## Shared authenticated state within micro-frontend applications
+
+To share a library, you need to configure the library in the `webpack.config.js` to share. Let's start with `shell`. Open `projects/shell/src/webpack.config.js`.
```

Comment:
do you have to say "to share" twice here? or does this work: "To share a library, you need to configure the library in the `webpack.config.js`" 
I can't tell... :)